db/db_iter.h

```diff
@@ -124,6 +124,7 @@ class DBIter final : public Iterator {
   void operator=(const DBIter&) = delete;
 
   ~DBIter() override {
+    status_.PermitUncheckedError();
```

Comment:
We prefer not to call PermitUncheckedError() in the destructor. Is it possible to force check it when using the DBIter? Or changing the testing code to check it?